gm4_chairs/data/gm4_chairs/advancements/hit.json

```diff
@@ -0,0 +1,22 @@
+{
+  "criteria": {
+    "punch": {
+      "trigger": "minecraft:player_hurt_entity",
+      "conditions": {
+        "entity": [
+          {
+            "condition": "minecraft:entity_properties",
+            "entity": "this",
+            "predicate": {
+              "type": "minecraft:pig",
+              "nbt": "{Tags:[\"gm4_chairs\"]}"
```

Comment:
This would be a good place to use the `gm4_chairs` team and skip needing an NBT check (even though it will have marginal benefits).